abstractions/go/request_info.go

```diff
@@ -19,6 +21,48 @@ type RequestInfo struct {
 	options         map[string]MiddlewareOption
 }
 
+func NewRequestInfo() *RequestInfo {
+	return &RequestInfo{
+		URI:             u.URL{},
+		Headers:         make(map[string]string),
+		QueryParameters: make(map[string]string),
+		options:         make(map[string]MiddlewareOption),
+	}
+}
+
+func (request *RequestInfo) SetUri(currentPath string, pathSegment string, isRawUrl bool) error {
+	if isRawUrl {
+		if currentPath == "" {
+			return errors.New("current path cannot be empty")
+		}
+		questionMarkSplat := strings.Split(currentPath, "?")
+		schemeHostAndPath := questionMarkSplat[0]
+		uri, err := url.Parse(schemeHostAndPath)
+		if err != nil {
+			return err
+		}
+		request.URI = *uri
+		if len(questionMarkSplat) > 1 {
+			queryParameters := questionMarkSplat[1]
+			for _, queryParameter := range strings.Split(queryParameters, "&") {
+				keyValue := strings.Split(queryParameter, "=")
+				if len(keyValue) == 2 {
+					request.QueryParameters[keyValue[0]] = keyValue[1]
+				} else if len(keyValue) == 1 {
+					request.QueryParameters[keyValue[0]] = ""
```

Comment:
Is there value in adding the query parameter in the event that the value is absent/null? (I think this applies to all the languages)

Reply:
I thought [this explanation](https://stackoverflow.com/a/4557490/3808675) was the best  🤣

Reply:
🤣 🤣 🤣 
If we can, we should probably remember to upvote that answer